_meta/beat.yml

```diff
@@ -247,6 +239,82 @@ apm-server:
     # Configure curve types for ECDHE based cipher suites.
     #ssl.curve_types: []
 
+  #---------------------------- APM Server - ILM Index Lifecycle Management ----------------------------
+
+  # Index Lifecycle Mangement (ILM)
+  #ilm:
+    # `enabled: "auto"`
+    # If a different output than Elasticsearch is configured, ILM will be disabled.
+    # If Elasticsearch output is configured, but specific `index` or `indices` settings are configured, ILM will be
+    # disabled, as it only works with default index settings.
+    # If the configured Elasticsearch instance is not eligible for ILM, ILM will also be disabled.
+    # If all preconditions are met, ILM will be enabled.
+    #
+    # `enabled: "true"`
+    # APM Server ignores any configured index settings. The configured output must be set to Elasticsearch and the
+    # instance needs to support ILM. Otherwise APM Server cannot apply ILM and creates unmanaged indices.
+    #
+    # `enabled: "false"`
+    # ILM is disabled and the server creates unmanaged indices.
+    #
+    # Default value is `auto`.
+    #enabled: "auto"
+
+    # Set `overwrite` to true for updating already existing ILM setup, including policies and templates. Default value is `false`.
+    #overwrite: false
+
+    # Set `require_policy` to `false` when policies are set up outside of APM Server but referenced here.
+    # Default value is `true`.
+    #require_policy: true
+
+    # The configured event types and policies will be merged with the default policies.
+    #mapping:
+      #- event_type: "error"
+      #  policy: "rollover-1-day"
+      #- event_type: "span"
+      #  policy: "rollover-1-day"
+      #- event_type: "transaction"
+      #  policy: "rollover-7-days"
+      #- event_type: "metric"
+      #  policy: "rollover-7-days"
+
+    # Configured policies are added to all pre-defined default policies.
+    # If a policy with the same name as a default policy is configured, the configured policy overwrites the default policy.
+    #policies:
+      #rollover-7-days:
```

Comment:
mixing values as keys in config dictionaries is always quite anti-intuitive to me, 
and we don't gain much compared with traditional `"name": "rollover-7-days"` approach. 
Elasticsearch does this a lot as well so i guess its ok... but I believe this is the only case of this in our own configuration, which makes it a bit awkward and not consistent. For instance, `indices` config reads like
```
  - index: "apm-%{[observer.version]}-error-%{+yyyy.MM.dd}"
      when.contains:
        processor.event: "error"
```

and not like

```
- "apm-%{[observer.version]}-error-%{+yyyy.MM.dd}"
      when.contains:
        processor.event: "error"
```

Reply:
We discussed removing `name` and `body` in this case and using the name as key, during one of our offline meetings to keep the config as short as possible. Therfore I changed to this format.
@graphaelli are you also in favor of changing back and introducing a `name` and `policy` approach again?

Reply:
i think removing `body` is a good idea either case

Reply:
(unrelated) i also think that default policy names should start with `apm-` prefix just like index names

Reply:
I got rid of it alltogether.